// @azure/ai-language-text/dist-esm/src/models.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
import { KnownErrorCode, KnownInnerErrorCode, } from "./generated";
/**
 * Type of actions supported by the {@link TextAnalysisClient.analyze} method.
 */
export const AnalyzeActionNames = {
    EntityLinking: "EntityLinking",
    EntityRecognition: "EntityRecognition",
    KeyPhraseExtraction: "KeyPhraseExtraction",
    PiiEntityRecognition: "PiiEntityRecognition",
    LanguageDetection: "LanguageDetection",
    SentimentAnalysis: "SentimentAnalysis",
    DynamicClassification: "DynamicClassification",
};
/**
 * Type of actions supported by the {@link TextAnalysisClient.beginAnalyzeBatch} method.
 */
export const AnalyzeBatchActionNames = {
    SentimentAnalysis: "SentimentAnalysis",
    EntityRecognition: "EntityRecognition",
    PiiEntityRecognition: "PiiEntityRecognition",
    KeyPhraseExtraction: "KeyPhraseExtraction",
    EntityLinking: "EntityLinking",
    Healthcare: "Healthcare",
    ExtractiveSummarization: "ExtractiveSummarization",
    AbstractiveSummarization: "AbstractiveSummarization",
    CustomEntityRecognition: "CustomEntityRecognition",
    CustomSingleLabelClassification: "CustomSingleLabelClassification",
    CustomMultiLabelClassification: "CustomMultiLabelClassification",
};
/**
 * Known values of the {@link HealthcareAction.fhirVersion} parameter.
 */
export var KnownFhirVersion;
(function (KnownFhirVersion) {
    /** 4.0.1 */
    KnownFhirVersion["4.0.1"] = "4.0.1";
})(KnownFhirVersion || (KnownFhirVersion = {}));
/**
 * Enum of possible error codes of a {@link TextAnalysisError}.
 */
export const KnownTextAnalysisErrorCode = Object.assign(Object.assign({}, KnownErrorCode), KnownInnerErrorCode);
