// @azure/ai-language-text/dist-esm/src/generated/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AnalyzeTextTaskKind} that the service accepts. */
export var KnownAnalyzeTextTaskKind;
(function (KnownAnalyzeTextTaskKind) {
    /** SentimentAnalysis */
    KnownAnalyzeTextTaskKind["SentimentAnalysis"] = "SentimentAnalysis";
    /** EntityRecognition */
    KnownAnalyzeTextTaskKind["EntityRecognition"] = "EntityRecognition";
    /** PiiEntityRecognition */
    KnownAnalyzeTextTaskKind["PiiEntityRecognition"] = "PiiEntityRecognition";
    /** KeyPhraseExtraction */
    KnownAnalyzeTextTaskKind["KeyPhraseExtraction"] = "KeyPhraseExtraction";
    /** LanguageDetection */
    KnownAnalyzeTextTaskKind["LanguageDetection"] = "LanguageDetection";
    /** EntityLinking */
    KnownAnalyzeTextTaskKind["EntityLinking"] = "EntityLinking";
    /** DynamicClassification */
    KnownAnalyzeTextTaskKind["DynamicClassification"] = "DynamicClassification";
})(KnownAnalyzeTextTaskKind || (KnownAnalyzeTextTaskKind = {}));
/** Known values of {@link AnalyzeTextTaskResultsKind} that the service accepts. */
export var KnownAnalyzeTextTaskResultsKind;
(function (KnownAnalyzeTextTaskResultsKind) {
    /** SentimentAnalysisResults */
    KnownAnalyzeTextTaskResultsKind["SentimentAnalysisResults"] = "SentimentAnalysisResults";
    /** EntityRecognitionResults */
    KnownAnalyzeTextTaskResultsKind["EntityRecognitionResults"] = "EntityRecognitionResults";
    /** PiiEntityRecognitionResults */
    KnownAnalyzeTextTaskResultsKind["PiiEntityRecognitionResults"] = "PiiEntityRecognitionResults";
    /** KeyPhraseExtractionResults */
    KnownAnalyzeTextTaskResultsKind["KeyPhraseExtractionResults"] = "KeyPhraseExtractionResults";
    /** LanguageDetectionResults */
    KnownAnalyzeTextTaskResultsKind["LanguageDetectionResults"] = "LanguageDetectionResults";
    /** EntityLinkingResults */
    KnownAnalyzeTextTaskResultsKind["EntityLinkingResults"] = "EntityLinkingResults";
    /** DynamicClassificationResults */
    KnownAnalyzeTextTaskResultsKind["DynamicClassificationResults"] = "DynamicClassificationResults";
})(KnownAnalyzeTextTaskResultsKind || (KnownAnalyzeTextTaskResultsKind = {}));
/** Known values of {@link ErrorCode} that the service accepts. */
export var KnownErrorCode;
(function (KnownErrorCode) {
    /** InvalidRequest */
    KnownErrorCode["InvalidRequest"] = "InvalidRequest";
    /** InvalidArgument */
    KnownErrorCode["InvalidArgument"] = "InvalidArgument";
    /** Unauthorized */
    KnownErrorCode["Unauthorized"] = "Unauthorized";
    /** Forbidden */
    KnownErrorCode["Forbidden"] = "Forbidden";
    /** NotFound */
    KnownErrorCode["NotFound"] = "NotFound";
    /** ProjectNotFound */
    KnownErrorCode["ProjectNotFound"] = "ProjectNotFound";
    /** OperationNotFound */
    KnownErrorCode["OperationNotFound"] = "OperationNotFound";
    /** AzureCognitiveSearchNotFound */
    KnownErrorCode["AzureCognitiveSearchNotFound"] = "AzureCognitiveSearchNotFound";
    /** AzureCognitiveSearchIndexNotFound */
    KnownErrorCode["AzureCognitiveSearchIndexNotFound"] = "AzureCognitiveSearchIndexNotFound";
    /** TooManyRequests */
    KnownErrorCode["TooManyRequests"] = "TooManyRequests";
    /** AzureCognitiveSearchThrottling */
    KnownErrorCode["AzureCognitiveSearchThrottling"] = "AzureCognitiveSearchThrottling";
    /** AzureCognitiveSearchIndexLimitReached */
    KnownErrorCode["AzureCognitiveSearchIndexLimitReached"] = "AzureCognitiveSearchIndexLimitReached";
    /** InternalServerError */
    KnownErrorCode["InternalServerError"] = "InternalServerError";
    /** ServiceUnavailable */
    KnownErrorCode["ServiceUnavailable"] = "ServiceUnavailable";
    /** Timeout */
    KnownErrorCode["Timeout"] = "Timeout";
    /** QuotaExceeded */
    KnownErrorCode["QuotaExceeded"] = "QuotaExceeded";
    /** Conflict */
    KnownErrorCode["Conflict"] = "Conflict";
    /** Warning */
    KnownErrorCode["Warning"] = "Warning";
})(KnownErrorCode || (KnownErrorCode = {}));
/** Known values of {@link InnerErrorCode} that the service accepts. */
export var KnownInnerErrorCode;
(function (KnownInnerErrorCode) {
    /** InvalidRequest */
    KnownInnerErrorCode["InvalidRequest"] = "InvalidRequest";
    /** InvalidParameterValue */
    KnownInnerErrorCode["InvalidParameterValue"] = "InvalidParameterValue";
    /** KnowledgeBaseNotFound */
    KnownInnerErrorCode["KnowledgeBaseNotFound"] = "KnowledgeBaseNotFound";
    /** AzureCognitiveSearchNotFound */
    KnownInnerErrorCode["AzureCognitiveSearchNotFound"] = "AzureCognitiveSearchNotFound";
    /** AzureCognitiveSearchThrottling */
    KnownInnerErrorCode["AzureCognitiveSearchThrottling"] = "AzureCognitiveSearchThrottling";
    /** ExtractionFailure */
    KnownInnerErrorCode["ExtractionFailure"] = "ExtractionFailure";
    /** InvalidRequestBodyFormat */
    KnownInnerErrorCode["InvalidRequestBodyFormat"] = "InvalidRequestBodyFormat";
    /** EmptyRequest */
    KnownInnerErrorCode["EmptyRequest"] = "EmptyRequest";
    /** MissingInputDocuments */
    KnownInnerErrorCode["MissingInputDocuments"] = "MissingInputDocuments";
    /** InvalidDocument */
    KnownInnerErrorCode["InvalidDocument"] = "InvalidDocument";
    /** ModelVersionIncorrect */
    KnownInnerErrorCode["ModelVersionIncorrect"] = "ModelVersionIncorrect";
    /** InvalidDocumentBatch */
    KnownInnerErrorCode["InvalidDocumentBatch"] = "InvalidDocumentBatch";
    /** UnsupportedLanguageCode */
    KnownInnerErrorCode["UnsupportedLanguageCode"] = "UnsupportedLanguageCode";
    /** InvalidCountryHint */
    KnownInnerErrorCode["InvalidCountryHint"] = "InvalidCountryHint";
})(KnownInnerErrorCode || (KnownInnerErrorCode = {}));
/** Known values of {@link AnalyzeTextLROTaskKind} that the service accepts. */
export var KnownAnalyzeTextLROTaskKind;
(function (KnownAnalyzeTextLROTaskKind) {
    /** SentimentAnalysis */
    KnownAnalyzeTextLROTaskKind["SentimentAnalysis"] = "SentimentAnalysis";
    /** EntityRecognition */
    KnownAnalyzeTextLROTaskKind["EntityRecognition"] = "EntityRecognition";
    /** PiiEntityRecognition */
    KnownAnalyzeTextLROTaskKind["PiiEntityRecognition"] = "PiiEntityRecognition";
    /** KeyPhraseExtraction */
    KnownAnalyzeTextLROTaskKind["KeyPhraseExtraction"] = "KeyPhraseExtraction";
    /** EntityLinking */
    KnownAnalyzeTextLROTaskKind["EntityLinking"] = "EntityLinking";
    /** Healthcare */
    KnownAnalyzeTextLROTaskKind["Healthcare"] = "Healthcare";
    /** ExtractiveSummarization */
    KnownAnalyzeTextLROTaskKind["ExtractiveSummarization"] = "ExtractiveSummarization";
    /** CustomEntityRecognition */
    KnownAnalyzeTextLROTaskKind["CustomEntityRecognition"] = "CustomEntityRecognition";
    /** CustomSingleLabelClassification */
    KnownAnalyzeTextLROTaskKind["CustomSingleLabelClassification"] = "CustomSingleLabelClassification";
    /** CustomMultiLabelClassification */
    KnownAnalyzeTextLROTaskKind["CustomMultiLabelClassification"] = "CustomMultiLabelClassification";
    /** AbstractiveSummarization */
    KnownAnalyzeTextLROTaskKind["AbstractiveSummarization"] = "AbstractiveSummarization";
})(KnownAnalyzeTextLROTaskKind || (KnownAnalyzeTextLROTaskKind = {}));
/** Known values of {@link OperationStatus} that the service accepts. */
export var KnownOperationStatus;
(function (KnownOperationStatus) {
    /** NotStarted */
    KnownOperationStatus["NotStarted"] = "notStarted";
    /** Running */
    KnownOperationStatus["Running"] = "running";
    /** Succeeded */
    KnownOperationStatus["Succeeded"] = "succeeded";
    /** PartiallyCompleted */
    KnownOperationStatus["PartiallyCompleted"] = "partiallyCompleted";
    /** Failed */
    KnownOperationStatus["Failed"] = "failed";
    /** Cancelled */
    KnownOperationStatus["Cancelled"] = "cancelled";
    /** Cancelling */
    KnownOperationStatus["Cancelling"] = "cancelling";
})(KnownOperationStatus || (KnownOperationStatus = {}));
/** Known values of {@link AnalyzeTextLROResultsKind} that the service accepts. */
export var KnownAnalyzeTextLROResultsKind;
(function (KnownAnalyzeTextLROResultsKind) {
    /** SentimentAnalysisLROResults */
    KnownAnalyzeTextLROResultsKind["SentimentAnalysisLROResults"] = "SentimentAnalysisLROResults";
    /** EntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["EntityRecognitionLROResults"] = "EntityRecognitionLROResults";
    /** PiiEntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["PiiEntityRecognitionLROResults"] = "PiiEntityRecognitionLROResults";
    /** KeyPhraseExtractionLROResults */
    KnownAnalyzeTextLROResultsKind["KeyPhraseExtractionLROResults"] = "KeyPhraseExtractionLROResults";
    /** EntityLinkingLROResults */
    KnownAnalyzeTextLROResultsKind["EntityLinkingLROResults"] = "EntityLinkingLROResults";
    /** HealthcareLROResults */
    KnownAnalyzeTextLROResultsKind["HealthcareLROResults"] = "HealthcareLROResults";
    /** ExtractiveSummarizationLROResults */
    KnownAnalyzeTextLROResultsKind["ExtractiveSummarizationLROResults"] = "ExtractiveSummarizationLROResults";
    /** CustomEntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["CustomEntityRecognitionLROResults"] = "CustomEntityRecognitionLROResults";
    /** CustomSingleLabelClassificationLROResults */
    KnownAnalyzeTextLROResultsKind["CustomSingleLabelClassificationLROResults"] = "CustomSingleLabelClassificationLROResults";
    /** CustomMultiLabelClassificationLROResults */
    KnownAnalyzeTextLROResultsKind["CustomMultiLabelClassificationLROResults"] = "CustomMultiLabelClassificationLROResults";
    /** AbstractiveSummarizationLROResults */
    KnownAnalyzeTextLROResultsKind["AbstractiveSummarizationLROResults"] = "AbstractiveSummarizationLROResults";
})(KnownAnalyzeTextLROResultsKind || (KnownAnalyzeTextLROResultsKind = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** NotStarted */
    KnownState["NotStarted"] = "notStarted";
    /** Running */
    KnownState["Running"] = "running";
    /** Succeeded */
    KnownState["Succeeded"] = "succeeded";
    /** Failed */
    KnownState["Failed"] = "failed";
    /** Cancelled */
    KnownState["Cancelled"] = "cancelled";
    /** Cancelling */
    KnownState["Cancelling"] = "cancelling";
})(KnownState || (KnownState = {}));
/** Known values of {@link StringIndexType} that the service accepts. */
export var KnownStringIndexType;
(function (KnownStringIndexType) {
    /** Returned offset and length values will correspond to TextElements (Graphemes and Grapheme clusters) confirming to the Unicode 8.0.0 standard. Use this option if your application is written in .Net Framework or .Net Core and you will be using StringInfo. */
    KnownStringIndexType["TextElementsV8"] = "TextElements_v8";
    /** Returned offset and length values will correspond to Unicode code points. Use this option if your application is written in a language that support Unicode, for example Python. */
    KnownStringIndexType["UnicodeCodePoint"] = "UnicodeCodePoint";
    /** Returned offset and length values will correspond to UTF-16 code units. Use this option if your application is written in a language that support Unicode, for example Java, JavaScript. */
    KnownStringIndexType["Utf16CodeUnit"] = "Utf16CodeUnit";
})(KnownStringIndexType || (KnownStringIndexType = {}));
/** Known values of {@link PiiEntityDomain} that the service accepts. */
export var KnownPiiEntityDomain;
(function (KnownPiiEntityDomain) {
    /** Indicates that entities in the Personal Health Information domain should be redacted. */
    KnownPiiEntityDomain["Phi"] = "phi";
    /** Indicates that no domain is specified. */
    KnownPiiEntityDomain["None"] = "none";
})(KnownPiiEntityDomain || (KnownPiiEntityDomain = {}));
/** Known values of {@link PiiEntityCategory} that the service accepts. */
export var KnownPiiEntityCategory;
(function (KnownPiiEntityCategory) {
    /** ABARoutingNumber */
    KnownPiiEntityCategory["ABARoutingNumber"] = "ABARoutingNumber";
    /** ARNationalIdentityNumber */
    KnownPiiEntityCategory["ARNationalIdentityNumber"] = "ARNationalIdentityNumber";
    /** AUBankAccountNumber */
    KnownPiiEntityCategory["AUBankAccountNumber"] = "AUBankAccountNumber";
    /** AUDriversLicenseNumber */
    KnownPiiEntityCategory["AUDriversLicenseNumber"] = "AUDriversLicenseNumber";
    /** AUMedicalAccountNumber */
    KnownPiiEntityCategory["AUMedicalAccountNumber"] = "AUMedicalAccountNumber";
    /** AUPassportNumber */
    KnownPiiEntityCategory["AUPassportNumber"] = "AUPassportNumber";
    /** AUTaxFileNumber */
    KnownPiiEntityCategory["AUTaxFileNumber"] = "AUTaxFileNumber";
    /** AUBusinessNumber */
    KnownPiiEntityCategory["AUBusinessNumber"] = "AUBusinessNumber";
    /** AUCompanyNumber */
    KnownPiiEntityCategory["AUCompanyNumber"] = "AUCompanyNumber";
    /** ATIdentityCard */
    KnownPiiEntityCategory["ATIdentityCard"] = "ATIdentityCard";
    /** ATTaxIdentificationNumber */
    KnownPiiEntityCategory["ATTaxIdentificationNumber"] = "ATTaxIdentificationNumber";
    /** ATValueAddedTaxNumber */
    KnownPiiEntityCategory["ATValueAddedTaxNumber"] = "ATValueAddedTaxNumber";
    /** AzureDocumentDBAuthKey */
    KnownPiiEntityCategory["AzureDocumentDBAuthKey"] = "AzureDocumentDBAuthKey";
    /** AzureIaasDatabaseConnectionAndSQLString */
    KnownPiiEntityCategory["AzureIaasDatabaseConnectionAndSQLString"] = "AzureIAASDatabaseConnectionAndSQLString";
    /** AzureIoTConnectionString */
    KnownPiiEntityCategory["AzureIoTConnectionString"] = "AzureIoTConnectionString";
    /** AzurePublishSettingPassword */
    KnownPiiEntityCategory["AzurePublishSettingPassword"] = "AzurePublishSettingPassword";
    /** AzureRedisCacheString */
    KnownPiiEntityCategory["AzureRedisCacheString"] = "AzureRedisCacheString";
    /** AzureSAS */
    KnownPiiEntityCategory["AzureSAS"] = "AzureSAS";
    /** AzureServiceBusString */
    KnownPiiEntityCategory["AzureServiceBusString"] = "AzureServiceBusString";
    /** AzureStorageAccountKey */
    KnownPiiEntityCategory["AzureStorageAccountKey"] = "AzureStorageAccountKey";
    /** AzureStorageAccountGeneric */
    KnownPiiEntityCategory["AzureStorageAccountGeneric"] = "AzureStorageAccountGeneric";
    /** BENationalNumber */
    KnownPiiEntityCategory["BENationalNumber"] = "BENationalNumber";
    /** BENationalNumberV2 */
    KnownPiiEntityCategory["BENationalNumberV2"] = "BENationalNumberV2";
    /** BEValueAddedTaxNumber */
    KnownPiiEntityCategory["BEValueAddedTaxNumber"] = "BEValueAddedTaxNumber";
    /** BrcpfNumber */
    KnownPiiEntityCategory["BrcpfNumber"] = "BRCPFNumber";
    /** BRLegalEntityNumber */
    KnownPiiEntityCategory["BRLegalEntityNumber"] = "BRLegalEntityNumber";
    /** BRNationalIdrg */
    KnownPiiEntityCategory["BRNationalIdrg"] = "BRNationalIDRG";
    /** BGUniformCivilNumber */
    KnownPiiEntityCategory["BGUniformCivilNumber"] = "BGUniformCivilNumber";
    /** CABankAccountNumber */
    KnownPiiEntityCategory["CABankAccountNumber"] = "CABankAccountNumber";
    /** CADriversLicenseNumber */
    KnownPiiEntityCategory["CADriversLicenseNumber"] = "CADriversLicenseNumber";
    /** CAHealthServiceNumber */
    KnownPiiEntityCategory["CAHealthServiceNumber"] = "CAHealthServiceNumber";
    /** CAPassportNumber */
    KnownPiiEntityCategory["CAPassportNumber"] = "CAPassportNumber";
    /** CAPersonalHealthIdentification */
    KnownPiiEntityCategory["CAPersonalHealthIdentification"] = "CAPersonalHealthIdentification";
    /** CASocialInsuranceNumber */
    KnownPiiEntityCategory["CASocialInsuranceNumber"] = "CASocialInsuranceNumber";
    /** CLIdentityCardNumber */
    KnownPiiEntityCategory["CLIdentityCardNumber"] = "CLIdentityCardNumber";
    /** CNResidentIdentityCardNumber */
    KnownPiiEntityCategory["CNResidentIdentityCardNumber"] = "CNResidentIdentityCardNumber";
    /** CreditCardNumber */
    KnownPiiEntityCategory["CreditCardNumber"] = "CreditCardNumber";
    /** HRIdentityCardNumber */
    KnownPiiEntityCategory["HRIdentityCardNumber"] = "HRIdentityCardNumber";
    /** HRNationalIDNumber */
    KnownPiiEntityCategory["HRNationalIDNumber"] = "HRNationalIDNumber";
    /** HRPersonalIdentificationNumber */
    KnownPiiEntityCategory["HRPersonalIdentificationNumber"] = "HRPersonalIdentificationNumber";
    /** HRPersonalIdentificationOIBNumberV2 */
    KnownPiiEntityCategory["HRPersonalIdentificationOIBNumberV2"] = "HRPersonalIdentificationOIBNumberV2";
    /** CYIdentityCard */
    KnownPiiEntityCategory["CYIdentityCard"] = "CYIdentityCard";
    /** CYTaxIdentificationNumber */
    KnownPiiEntityCategory["CYTaxIdentificationNumber"] = "CYTaxIdentificationNumber";
    /** CZPersonalIdentityNumber */
    KnownPiiEntityCategory["CZPersonalIdentityNumber"] = "CZPersonalIdentityNumber";
    /** CZPersonalIdentityV2 */
    KnownPiiEntityCategory["CZPersonalIdentityV2"] = "CZPersonalIdentityV2";
    /** DKPersonalIdentificationNumber */
    KnownPiiEntityCategory["DKPersonalIdentificationNumber"] = "DKPersonalIdentificationNumber";
    /** DKPersonalIdentificationV2 */
    KnownPiiEntityCategory["DKPersonalIdentificationV2"] = "DKPersonalIdentificationV2";
    /** DrugEnforcementAgencyNumber */
    KnownPiiEntityCategory["DrugEnforcementAgencyNumber"] = "DrugEnforcementAgencyNumber";
    /** EEPersonalIdentificationCode */
    KnownPiiEntityCategory["EEPersonalIdentificationCode"] = "EEPersonalIdentificationCode";
    /** EUDebitCardNumber */
    KnownPiiEntityCategory["EUDebitCardNumber"] = "EUDebitCardNumber";
    /** EUDriversLicenseNumber */
    KnownPiiEntityCategory["EUDriversLicenseNumber"] = "EUDriversLicenseNumber";
    /** EugpsCoordinates */
    KnownPiiEntityCategory["EugpsCoordinates"] = "EUGPSCoordinates";
    /** EUNationalIdentificationNumber */
    KnownPiiEntityCategory["EUNationalIdentificationNumber"] = "EUNationalIdentificationNumber";
    /** EUPassportNumber */
    KnownPiiEntityCategory["EUPassportNumber"] = "EUPassportNumber";
    /** EUSocialSecurityNumber */
    KnownPiiEntityCategory["EUSocialSecurityNumber"] = "EUSocialSecurityNumber";
    /** EUTaxIdentificationNumber */
    KnownPiiEntityCategory["EUTaxIdentificationNumber"] = "EUTaxIdentificationNumber";
    /** FIEuropeanHealthNumber */
    KnownPiiEntityCategory["FIEuropeanHealthNumber"] = "FIEuropeanHealthNumber";
    /** FINationalID */
    KnownPiiEntityCategory["FINationalID"] = "FINationalID";
    /** FINationalIDV2 */
    KnownPiiEntityCategory["FINationalIDV2"] = "FINationalIDV2";
    /** FIPassportNumber */
    KnownPiiEntityCategory["FIPassportNumber"] = "FIPassportNumber";
    /** FRDriversLicenseNumber */
    KnownPiiEntityCategory["FRDriversLicenseNumber"] = "FRDriversLicenseNumber";
    /** FRHealthInsuranceNumber */
    KnownPiiEntityCategory["FRHealthInsuranceNumber"] = "FRHealthInsuranceNumber";
    /** FRNationalID */
    KnownPiiEntityCategory["FRNationalID"] = "FRNationalID";
    /** FRPassportNumber */
    KnownPiiEntityCategory["FRPassportNumber"] = "FRPassportNumber";
    /** FRSocialSecurityNumber */
    KnownPiiEntityCategory["FRSocialSecurityNumber"] = "FRSocialSecurityNumber";
    /** FRTaxIdentificationNumber */
    KnownPiiEntityCategory["FRTaxIdentificationNumber"] = "FRTaxIdentificationNumber";
    /** FRValueAddedTaxNumber */
    KnownPiiEntityCategory["FRValueAddedTaxNumber"] = "FRValueAddedTaxNumber";
    /** DEDriversLicenseNumber */
    KnownPiiEntityCategory["DEDriversLicenseNumber"] = "DEDriversLicenseNumber";
    /** DEPassportNumber */
    KnownPiiEntityCategory["DEPassportNumber"] = "DEPassportNumber";
    /** DEIdentityCardNumber */
    KnownPiiEntityCategory["DEIdentityCardNumber"] = "DEIdentityCardNumber";
    /** DETaxIdentificationNumber */
    KnownPiiEntityCategory["DETaxIdentificationNumber"] = "DETaxIdentificationNumber";
    /** DEValueAddedNumber */
    KnownPiiEntityCategory["DEValueAddedNumber"] = "DEValueAddedNumber";
    /** GRNationalIDCard */
    KnownPiiEntityCategory["GRNationalIDCard"] = "GRNationalIDCard";
    /** GRNationalIDV2 */
    KnownPiiEntityCategory["GRNationalIDV2"] = "GRNationalIDV2";
    /** GRTaxIdentificationNumber */
    KnownPiiEntityCategory["GRTaxIdentificationNumber"] = "GRTaxIdentificationNumber";
    /** HKIdentityCardNumber */
    KnownPiiEntityCategory["HKIdentityCardNumber"] = "HKIdentityCardNumber";
    /** HUValueAddedNumber */
    KnownPiiEntityCategory["HUValueAddedNumber"] = "HUValueAddedNumber";
    /** HUPersonalIdentificationNumber */
    KnownPiiEntityCategory["HUPersonalIdentificationNumber"] = "HUPersonalIdentificationNumber";
    /** HUTaxIdentificationNumber */
    KnownPiiEntityCategory["HUTaxIdentificationNumber"] = "HUTaxIdentificationNumber";
    /** INPermanentAccount */
    KnownPiiEntityCategory["INPermanentAccount"] = "INPermanentAccount";
    /** INUniqueIdentificationNumber */
    KnownPiiEntityCategory["INUniqueIdentificationNumber"] = "INUniqueIdentificationNumber";
    /** IDIdentityCardNumber */
    KnownPiiEntityCategory["IDIdentityCardNumber"] = "IDIdentityCardNumber";
    /** InternationalBankingAccountNumber */
    KnownPiiEntityCategory["InternationalBankingAccountNumber"] = "InternationalBankingAccountNumber";
    /** IEPersonalPublicServiceNumber */
    KnownPiiEntityCategory["IEPersonalPublicServiceNumber"] = "IEPersonalPublicServiceNumber";
    /** IEPersonalPublicServiceNumberV2 */
    KnownPiiEntityCategory["IEPersonalPublicServiceNumberV2"] = "IEPersonalPublicServiceNumberV2";
    /** ILBankAccountNumber */
    KnownPiiEntityCategory["ILBankAccountNumber"] = "ILBankAccountNumber";
    /** ILNationalID */
    KnownPiiEntityCategory["ILNationalID"] = "ILNationalID";
    /** ITDriversLicenseNumber */
    KnownPiiEntityCategory["ITDriversLicenseNumber"] = "ITDriversLicenseNumber";
    /** ITFiscalCode */
    KnownPiiEntityCategory["ITFiscalCode"] = "ITFiscalCode";
    /** ITValueAddedTaxNumber */
    KnownPiiEntityCategory["ITValueAddedTaxNumber"] = "ITValueAddedTaxNumber";
    /** JPBankAccountNumber */
    KnownPiiEntityCategory["JPBankAccountNumber"] = "JPBankAccountNumber";
    /** JPDriversLicenseNumber */
    KnownPiiEntityCategory["JPDriversLicenseNumber"] = "JPDriversLicenseNumber";
    /** JPPassportNumber */
    KnownPiiEntityCategory["JPPassportNumber"] = "JPPassportNumber";
    /** JPResidentRegistrationNumber */
    KnownPiiEntityCategory["JPResidentRegistrationNumber"] = "JPResidentRegistrationNumber";
    /** JPSocialInsuranceNumber */
    KnownPiiEntityCategory["JPSocialInsuranceNumber"] = "JPSocialInsuranceNumber";
    /** JPMyNumberCorporate */
    KnownPiiEntityCategory["JPMyNumberCorporate"] = "JPMyNumberCorporate";
    /** JPMyNumberPersonal */
    KnownPiiEntityCategory["JPMyNumberPersonal"] = "JPMyNumberPersonal";
    /** JPResidenceCardNumber */
    KnownPiiEntityCategory["JPResidenceCardNumber"] = "JPResidenceCardNumber";
    /** LVPersonalCode */
    KnownPiiEntityCategory["LVPersonalCode"] = "LVPersonalCode";
    /** LTPersonalCode */
    KnownPiiEntityCategory["LTPersonalCode"] = "LTPersonalCode";
    /** LUNationalIdentificationNumberNatural */
    KnownPiiEntityCategory["LUNationalIdentificationNumberNatural"] = "LUNationalIdentificationNumberNatural";
    /** LUNationalIdentificationNumberNonNatural */
    KnownPiiEntityCategory["LUNationalIdentificationNumberNonNatural"] = "LUNationalIdentificationNumberNonNatural";
    /** MYIdentityCardNumber */
    KnownPiiEntityCategory["MYIdentityCardNumber"] = "MYIdentityCardNumber";
    /** MTIdentityCardNumber */
    KnownPiiEntityCategory["MTIdentityCardNumber"] = "MTIdentityCardNumber";
    /** MTTaxIDNumber */
    KnownPiiEntityCategory["MTTaxIDNumber"] = "MTTaxIDNumber";
    /** NLCitizensServiceNumber */
    KnownPiiEntityCategory["NLCitizensServiceNumber"] = "NLCitizensServiceNumber";
    /** NLCitizensServiceNumberV2 */
    KnownPiiEntityCategory["NLCitizensServiceNumberV2"] = "NLCitizensServiceNumberV2";
    /** NLTaxIdentificationNumber */
    KnownPiiEntityCategory["NLTaxIdentificationNumber"] = "NLTaxIdentificationNumber";
    /** NLValueAddedTaxNumber */
    KnownPiiEntityCategory["NLValueAddedTaxNumber"] = "NLValueAddedTaxNumber";
    /** NZBankAccountNumber */
    KnownPiiEntityCategory["NZBankAccountNumber"] = "NZBankAccountNumber";
    /** NZDriversLicenseNumber */
    KnownPiiEntityCategory["NZDriversLicenseNumber"] = "NZDriversLicenseNumber";
    /** NZInlandRevenueNumber */
    KnownPiiEntityCategory["NZInlandRevenueNumber"] = "NZInlandRevenueNumber";
    /** NZMinistryOfHealthNumber */
    KnownPiiEntityCategory["NZMinistryOfHealthNumber"] = "NZMinistryOfHealthNumber";
    /** NZSocialWelfareNumber */
    KnownPiiEntityCategory["NZSocialWelfareNumber"] = "NZSocialWelfareNumber";
    /** NOIdentityNumber */
    KnownPiiEntityCategory["NOIdentityNumber"] = "NOIdentityNumber";
    /** PHUnifiedMultiPurposeIDNumber */
    KnownPiiEntityCategory["PHUnifiedMultiPurposeIDNumber"] = "PHUnifiedMultiPurposeIDNumber";
    /** PLIdentityCard */
    KnownPiiEntityCategory["PLIdentityCard"] = "PLIdentityCard";
    /** PLNationalID */
    KnownPiiEntityCategory["PLNationalID"] = "PLNationalID";
    /** PLNationalIDV2 */
    KnownPiiEntityCategory["PLNationalIDV2"] = "PLNationalIDV2";
    /** PLPassportNumber */
    KnownPiiEntityCategory["PLPassportNumber"] = "PLPassportNumber";
    /** PLTaxIdentificationNumber */
    KnownPiiEntityCategory["PLTaxIdentificationNumber"] = "PLTaxIdentificationNumber";
    /** PlregonNumber */
    KnownPiiEntityCategory["PlregonNumber"] = "PLREGONNumber";
    /** PTCitizenCardNumber */
    KnownPiiEntityCategory["PTCitizenCardNumber"] = "PTCitizenCardNumber";
    /** PTCitizenCardNumberV2 */
    KnownPiiEntityCategory["PTCitizenCardNumberV2"] = "PTCitizenCardNumberV2";
    /** PTTaxIdentificationNumber */
    KnownPiiEntityCategory["PTTaxIdentificationNumber"] = "PTTaxIdentificationNumber";
    /** ROPersonalNumericalCode */
    KnownPiiEntityCategory["ROPersonalNumericalCode"] = "ROPersonalNumericalCode";
    /** RUPassportNumberDomestic */
    KnownPiiEntityCategory["RUPassportNumberDomestic"] = "RUPassportNumberDomestic";
    /** RUPassportNumberInternational */
    KnownPiiEntityCategory["RUPassportNumberInternational"] = "RUPassportNumberInternational";
    /** SANationalID */
    KnownPiiEntityCategory["SANationalID"] = "SANationalID";
    /** SGNationalRegistrationIdentityCardNumber */
    KnownPiiEntityCategory["SGNationalRegistrationIdentityCardNumber"] = "SGNationalRegistrationIdentityCardNumber";
    /** SKPersonalNumber */
    KnownPiiEntityCategory["SKPersonalNumber"] = "SKPersonalNumber";
    /** SITaxIdentificationNumber */
    KnownPiiEntityCategory["SITaxIdentificationNumber"] = "SITaxIdentificationNumber";
    /** SIUniqueMasterCitizenNumber */
    KnownPiiEntityCategory["SIUniqueMasterCitizenNumber"] = "SIUniqueMasterCitizenNumber";
    /** ZAIdentificationNumber */
    KnownPiiEntityCategory["ZAIdentificationNumber"] = "ZAIdentificationNumber";
    /** KRResidentRegistrationNumber */
    KnownPiiEntityCategory["KRResidentRegistrationNumber"] = "KRResidentRegistrationNumber";
    /** Esdni */
    KnownPiiEntityCategory["Esdni"] = "ESDNI";
    /** ESSocialSecurityNumber */
    KnownPiiEntityCategory["ESSocialSecurityNumber"] = "ESSocialSecurityNumber";
    /** ESTaxIdentificationNumber */
    KnownPiiEntityCategory["ESTaxIdentificationNumber"] = "ESTaxIdentificationNumber";
    /** SQLServerConnectionString */
    KnownPiiEntityCategory["SQLServerConnectionString"] = "SQLServerConnectionString";
    /** SENationalID */
    KnownPiiEntityCategory["SENationalID"] = "SENationalID";
    /** SENationalIDV2 */
    KnownPiiEntityCategory["SENationalIDV2"] = "SENationalIDV2";
    /** SEPassportNumber */
    KnownPiiEntityCategory["SEPassportNumber"] = "SEPassportNumber";
    /** SETaxIdentificationNumber */
    KnownPiiEntityCategory["SETaxIdentificationNumber"] = "SETaxIdentificationNumber";
    /** SwiftCode */
    KnownPiiEntityCategory["SwiftCode"] = "SWIFTCode";
    /** CHSocialSecurityNumber */
    KnownPiiEntityCategory["CHSocialSecurityNumber"] = "CHSocialSecurityNumber";
    /** TWNationalID */
    KnownPiiEntityCategory["TWNationalID"] = "TWNationalID";
    /** TWPassportNumber */
    KnownPiiEntityCategory["TWPassportNumber"] = "TWPassportNumber";
    /** TWResidentCertificate */
    KnownPiiEntityCategory["TWResidentCertificate"] = "TWResidentCertificate";
    /** THPopulationIdentificationCode */
    KnownPiiEntityCategory["THPopulationIdentificationCode"] = "THPopulationIdentificationCode";
    /** TRNationalIdentificationNumber */
    KnownPiiEntityCategory["TRNationalIdentificationNumber"] = "TRNationalIdentificationNumber";
    /** UKDriversLicenseNumber */
    KnownPiiEntityCategory["UKDriversLicenseNumber"] = "UKDriversLicenseNumber";
    /** UKElectoralRollNumber */
    KnownPiiEntityCategory["UKElectoralRollNumber"] = "UKElectoralRollNumber";
    /** UKNationalHealthNumber */
    KnownPiiEntityCategory["UKNationalHealthNumber"] = "UKNationalHealthNumber";
    /** UKNationalInsuranceNumber */
    KnownPiiEntityCategory["UKNationalInsuranceNumber"] = "UKNationalInsuranceNumber";
    /** UKUniqueTaxpayerNumber */
    KnownPiiEntityCategory["UKUniqueTaxpayerNumber"] = "UKUniqueTaxpayerNumber";
    /** UsukPassportNumber */
    KnownPiiEntityCategory["UsukPassportNumber"] = "USUKPassportNumber";
    /** USBankAccountNumber */
    KnownPiiEntityCategory["USBankAccountNumber"] = "USBankAccountNumber";
    /** USDriversLicenseNumber */
    KnownPiiEntityCategory["USDriversLicenseNumber"] = "USDriversLicenseNumber";
    /** USIndividualTaxpayerIdentification */
    KnownPiiEntityCategory["USIndividualTaxpayerIdentification"] = "USIndividualTaxpayerIdentification";
    /** USSocialSecurityNumber */
    KnownPiiEntityCategory["USSocialSecurityNumber"] = "USSocialSecurityNumber";
    /** UAPassportNumberDomestic */
    KnownPiiEntityCategory["UAPassportNumberDomestic"] = "UAPassportNumberDomestic";
    /** UAPassportNumberInternational */
    KnownPiiEntityCategory["UAPassportNumberInternational"] = "UAPassportNumberInternational";
    /** Organization */
    KnownPiiEntityCategory["Organization"] = "Organization";
    /** Email */
    KnownPiiEntityCategory["Email"] = "Email";
    /** URL */
    KnownPiiEntityCategory["URL"] = "URL";
    /** Age */
    KnownPiiEntityCategory["Age"] = "Age";
    /** PhoneNumber */
    KnownPiiEntityCategory["PhoneNumber"] = "PhoneNumber";
    /** IPAddress */
    KnownPiiEntityCategory["IPAddress"] = "IPAddress";
    /** Date */
    KnownPiiEntityCategory["Date"] = "Date";
    /** Person */
    KnownPiiEntityCategory["Person"] = "Person";
    /** Address */
    KnownPiiEntityCategory["Address"] = "Address";
    /** All */
    KnownPiiEntityCategory["All"] = "All";
    /** Default */
    KnownPiiEntityCategory["Default"] = "Default";
})(KnownPiiEntityCategory || (KnownPiiEntityCategory = {}));
/** Known values of {@link ClassificationType} that the service accepts. */
export var KnownClassificationType;
(function (KnownClassificationType) {
    /** Single */
    KnownClassificationType["Single"] = "Single";
    /** Multi */
    KnownClassificationType["Multi"] = "Multi";
})(KnownClassificationType || (KnownClassificationType = {}));
/** Known values of {@link ResolutionKind} that the service accepts. */
export var KnownResolutionKind;
(function (KnownResolutionKind) {
    /** BooleanResolution */
    KnownResolutionKind["BooleanResolution"] = "BooleanResolution";
    /** DateTimeResolution */
    KnownResolutionKind["DateTimeResolution"] = "DateTimeResolution";
    /** NumberResolution */
    KnownResolutionKind["NumberResolution"] = "NumberResolution";
    /** OrdinalResolution */
    KnownResolutionKind["OrdinalResolution"] = "OrdinalResolution";
    /** SpeedResolution */
    KnownResolutionKind["SpeedResolution"] = "SpeedResolution";
    /** WeightResolution */
    KnownResolutionKind["WeightResolution"] = "WeightResolution";
    /** LengthResolution */
    KnownResolutionKind["LengthResolution"] = "LengthResolution";
    /** VolumeResolution */
    KnownResolutionKind["VolumeResolution"] = "VolumeResolution";
    /** AreaResolution */
    KnownResolutionKind["AreaResolution"] = "AreaResolution";
    /** AgeResolution */
    KnownResolutionKind["AgeResolution"] = "AgeResolution";
    /** InformationResolution */
    KnownResolutionKind["InformationResolution"] = "InformationResolution";
    /** TemperatureResolution */
    KnownResolutionKind["TemperatureResolution"] = "TemperatureResolution";
    /** CurrencyResolution */
    KnownResolutionKind["CurrencyResolution"] = "CurrencyResolution";
    /** NumericRangeResolution */
    KnownResolutionKind["NumericRangeResolution"] = "NumericRangeResolution";
    /** TemporalSpanResolution */
    KnownResolutionKind["TemporalSpanResolution"] = "TemporalSpanResolution";
})(KnownResolutionKind || (KnownResolutionKind = {}));
/** Known values of {@link WarningCode} that the service accepts. */
export var KnownWarningCode;
(function (KnownWarningCode) {
    /** LongWordsInDocument */
    KnownWarningCode["LongWordsInDocument"] = "LongWordsInDocument";
    /** DocumentTruncated */
    KnownWarningCode["DocumentTruncated"] = "DocumentTruncated";
})(KnownWarningCode || (KnownWarningCode = {}));
/** Known values of {@link ScriptKind} that the service accepts. */
export var KnownScriptKind;
(function (KnownScriptKind) {
    /** Latin */
    KnownScriptKind["Latin"] = "Latin";
})(KnownScriptKind || (KnownScriptKind = {}));
/** Known values of {@link FhirVersion} that the service accepts. */
export var KnownFhirVersion;
(function (KnownFhirVersion) {
    /** Four01 */
    KnownFhirVersion["Four01"] = "4.0.1";
})(KnownFhirVersion || (KnownFhirVersion = {}));
/** Known values of {@link HealthcareDocumentType} that the service accepts. */
export var KnownHealthcareDocumentType;
(function (KnownHealthcareDocumentType) {
    /** None */
    KnownHealthcareDocumentType["None"] = "None";
    /** ClinicalTrial */
    KnownHealthcareDocumentType["ClinicalTrial"] = "ClinicalTrial";
    /** DischargeSummary */
    KnownHealthcareDocumentType["DischargeSummary"] = "DischargeSummary";
    /** ProgressNote */
    KnownHealthcareDocumentType["ProgressNote"] = "ProgressNote";
    /** HistoryAndPhysical */
    KnownHealthcareDocumentType["HistoryAndPhysical"] = "HistoryAndPhysical";
    /** Consult */
    KnownHealthcareDocumentType["Consult"] = "Consult";
    /** Imaging */
    KnownHealthcareDocumentType["Imaging"] = "Imaging";
    /** Pathology */
    KnownHealthcareDocumentType["Pathology"] = "Pathology";
    /** ProcedureNote */
    KnownHealthcareDocumentType["ProcedureNote"] = "ProcedureNote";
})(KnownHealthcareDocumentType || (KnownHealthcareDocumentType = {}));
/** Known values of {@link HealthcareEntityCategory} that the service accepts. */
export var KnownHealthcareEntityCategory;
(function (KnownHealthcareEntityCategory) {
    /** BodyStructure */
    KnownHealthcareEntityCategory["BodyStructure"] = "BodyStructure";
    /** Age */
    KnownHealthcareEntityCategory["Age"] = "Age";
    /** Gender */
    KnownHealthcareEntityCategory["Gender"] = "Gender";
    /** ExaminationName */
    KnownHealthcareEntityCategory["ExaminationName"] = "ExaminationName";
    /** Date */
    KnownHealthcareEntityCategory["Date"] = "Date";
    /** Direction */
    KnownHealthcareEntityCategory["Direction"] = "Direction";
    /** Frequency */
    KnownHealthcareEntityCategory["Frequency"] = "Frequency";
    /** MeasurementValue */
    KnownHealthcareEntityCategory["MeasurementValue"] = "MeasurementValue";
    /** MeasurementUnit */
    KnownHealthcareEntityCategory["MeasurementUnit"] = "MeasurementUnit";
    /** RelationalOperator */
    KnownHealthcareEntityCategory["RelationalOperator"] = "RelationalOperator";
    /** Time */
    KnownHealthcareEntityCategory["Time"] = "Time";
    /** GeneOrProtein */
    KnownHealthcareEntityCategory["GeneOrProtein"] = "GeneOrProtein";
    /** Variant */
    KnownHealthcareEntityCategory["Variant"] = "Variant";
    /** AdministrativeEvent */
    KnownHealthcareEntityCategory["AdministrativeEvent"] = "AdministrativeEvent";
    /** CareEnvironment */
    KnownHealthcareEntityCategory["CareEnvironment"] = "CareEnvironment";
    /** HealthcareProfession */
    KnownHealthcareEntityCategory["HealthcareProfession"] = "HealthcareProfession";
    /** Diagnosis */
    KnownHealthcareEntityCategory["Diagnosis"] = "Diagnosis";
    /** SymptomOrSign */
    KnownHealthcareEntityCategory["SymptomOrSign"] = "SymptomOrSign";
    /** ConditionQualifier */
    KnownHealthcareEntityCategory["ConditionQualifier"] = "ConditionQualifier";
    /** MedicationClass */
    KnownHealthcareEntityCategory["MedicationClass"] = "MedicationClass";
    /** MedicationName */
    KnownHealthcareEntityCategory["MedicationName"] = "MedicationName";
    /** Dosage */
    KnownHealthcareEntityCategory["Dosage"] = "Dosage";
    /** MedicationForm */
    KnownHealthcareEntityCategory["MedicationForm"] = "MedicationForm";
    /** MedicationRoute */
    KnownHealthcareEntityCategory["MedicationRoute"] = "MedicationRoute";
    /** FamilyRelation */
    KnownHealthcareEntityCategory["FamilyRelation"] = "FamilyRelation";
    /** TreatmentName */
    KnownHealthcareEntityCategory["TreatmentName"] = "TreatmentName";
})(KnownHealthcareEntityCategory || (KnownHealthcareEntityCategory = {}));
/** Known values of {@link RelationType} that the service accepts. */
export var KnownRelationType;
(function (KnownRelationType) {
    /** Abbreviation */
    KnownRelationType["Abbreviation"] = "Abbreviation";
    /** DirectionOfBodyStructure */
    KnownRelationType["DirectionOfBodyStructure"] = "DirectionOfBodyStructure";
    /** DirectionOfCondition */
    KnownRelationType["DirectionOfCondition"] = "DirectionOfCondition";
    /** DirectionOfExamination */
    KnownRelationType["DirectionOfExamination"] = "DirectionOfExamination";
    /** DirectionOfTreatment */
    KnownRelationType["DirectionOfTreatment"] = "DirectionOfTreatment";
    /** DosageOfMedication */
    KnownRelationType["DosageOfMedication"] = "DosageOfMedication";
    /** FormOfMedication */
    KnownRelationType["FormOfMedication"] = "FormOfMedication";
    /** FrequencyOfMedication */
    KnownRelationType["FrequencyOfMedication"] = "FrequencyOfMedication";
    /** FrequencyOfTreatment */
    KnownRelationType["FrequencyOfTreatment"] = "FrequencyOfTreatment";
    /** QualifierOfCondition */
    KnownRelationType["QualifierOfCondition"] = "QualifierOfCondition";
    /** RelationOfExamination */
    KnownRelationType["RelationOfExamination"] = "RelationOfExamination";
    /** RouteOfMedication */
    KnownRelationType["RouteOfMedication"] = "RouteOfMedication";
    /** TimeOfCondition */
    KnownRelationType["TimeOfCondition"] = "TimeOfCondition";
    /** TimeOfEvent */
    KnownRelationType["TimeOfEvent"] = "TimeOfEvent";
    /** TimeOfExamination */
    KnownRelationType["TimeOfExamination"] = "TimeOfExamination";
    /** TimeOfMedication */
    KnownRelationType["TimeOfMedication"] = "TimeOfMedication";
    /** TimeOfTreatment */
    KnownRelationType["TimeOfTreatment"] = "TimeOfTreatment";
    /** UnitOfCondition */
    KnownRelationType["UnitOfCondition"] = "UnitOfCondition";
    /** UnitOfExamination */
    KnownRelationType["UnitOfExamination"] = "UnitOfExamination";
    /** ValueOfCondition */
    KnownRelationType["ValueOfCondition"] = "ValueOfCondition";
    /** ValueOfExamination */
    KnownRelationType["ValueOfExamination"] = "ValueOfExamination";
})(KnownRelationType || (KnownRelationType = {}));
/** Known values of {@link AgeUnit} that the service accepts. */
export var KnownAgeUnit;
(function (KnownAgeUnit) {
    /** Unspecified */
    KnownAgeUnit["Unspecified"] = "Unspecified";
    /** Year */
    KnownAgeUnit["Year"] = "Year";
    /** Month */
    KnownAgeUnit["Month"] = "Month";
    /** Week */
    KnownAgeUnit["Week"] = "Week";
    /** Day */
    KnownAgeUnit["Day"] = "Day";
})(KnownAgeUnit || (KnownAgeUnit = {}));
/** Known values of {@link VolumeUnit} that the service accepts. */
export var KnownVolumeUnit;
(function (KnownVolumeUnit) {
    /** Unspecified */
    KnownVolumeUnit["Unspecified"] = "Unspecified";
    /** CubicMeter */
    KnownVolumeUnit["CubicMeter"] = "CubicMeter";
    /** CubicCentimeter */
    KnownVolumeUnit["CubicCentimeter"] = "CubicCentimeter";
    /** CubicMillimeter */
    KnownVolumeUnit["CubicMillimeter"] = "CubicMillimeter";
    /** Hectoliter */
    KnownVolumeUnit["Hectoliter"] = "Hectoliter";
    /** Decaliter */
    KnownVolumeUnit["Decaliter"] = "Decaliter";
    /** Liter */
    KnownVolumeUnit["Liter"] = "Liter";
    /** Centiliter */
    KnownVolumeUnit["Centiliter"] = "Centiliter";
    /** Milliliter */
    KnownVolumeUnit["Milliliter"] = "Milliliter";
    /** CubicYard */
    KnownVolumeUnit["CubicYard"] = "CubicYard";
    /** CubicInch */
    KnownVolumeUnit["CubicInch"] = "CubicInch";
    /** CubicFoot */
    KnownVolumeUnit["CubicFoot"] = "CubicFoot";
    /** CubicMile */
    KnownVolumeUnit["CubicMile"] = "CubicMile";
    /** FluidOunce */
    KnownVolumeUnit["FluidOunce"] = "FluidOunce";
    /** Teaspoon */
    KnownVolumeUnit["Teaspoon"] = "Teaspoon";
    /** Tablespoon */
    KnownVolumeUnit["Tablespoon"] = "Tablespoon";
    /** Pint */
    KnownVolumeUnit["Pint"] = "Pint";
    /** Quart */
    KnownVolumeUnit["Quart"] = "Quart";
    /** Cup */
    KnownVolumeUnit["Cup"] = "Cup";
    /** Gill */
    KnownVolumeUnit["Gill"] = "Gill";
    /** Pinch */
    KnownVolumeUnit["Pinch"] = "Pinch";
    /** FluidDram */
    KnownVolumeUnit["FluidDram"] = "FluidDram";
    /** Barrel */
    KnownVolumeUnit["Barrel"] = "Barrel";
    /** Minim */
    KnownVolumeUnit["Minim"] = "Minim";
    /** Cord */
    KnownVolumeUnit["Cord"] = "Cord";
    /** Peck */
    KnownVolumeUnit["Peck"] = "Peck";
    /** Bushel */
    KnownVolumeUnit["Bushel"] = "Bushel";
    /** Hogshead */
    KnownVolumeUnit["Hogshead"] = "Hogshead";
})(KnownVolumeUnit || (KnownVolumeUnit = {}));
/** Known values of {@link SpeedUnit} that the service accepts. */
export var KnownSpeedUnit;
(function (KnownSpeedUnit) {
    /** Unspecified */
    KnownSpeedUnit["Unspecified"] = "Unspecified";
    /** MeterPerSecond */
    KnownSpeedUnit["MeterPerSecond"] = "MeterPerSecond";
    /** KilometerPerHour */
    KnownSpeedUnit["KilometerPerHour"] = "KilometerPerHour";
    /** KilometerPerMinute */
    KnownSpeedUnit["KilometerPerMinute"] = "KilometerPerMinute";
    /** KilometerPerSecond */
    KnownSpeedUnit["KilometerPerSecond"] = "KilometerPerSecond";
    /** MilePerHour */
    KnownSpeedUnit["MilePerHour"] = "MilePerHour";
    /** Knot */
    KnownSpeedUnit["Knot"] = "Knot";
    /** FootPerSecond */
    KnownSpeedUnit["FootPerSecond"] = "FootPerSecond";
    /** FootPerMinute */
    KnownSpeedUnit["FootPerMinute"] = "FootPerMinute";
    /** YardPerMinute */
    KnownSpeedUnit["YardPerMinute"] = "YardPerMinute";
    /** YardPerSecond */
    KnownSpeedUnit["YardPerSecond"] = "YardPerSecond";
    /** MeterPerMillisecond */
    KnownSpeedUnit["MeterPerMillisecond"] = "MeterPerMillisecond";
    /** CentimeterPerMillisecond */
    KnownSpeedUnit["CentimeterPerMillisecond"] = "CentimeterPerMillisecond";
    /** KilometerPerMillisecond */
    KnownSpeedUnit["KilometerPerMillisecond"] = "KilometerPerMillisecond";
})(KnownSpeedUnit || (KnownSpeedUnit = {}));
/** Known values of {@link AreaUnit} that the service accepts. */
export var KnownAreaUnit;
(function (KnownAreaUnit) {
    /** Unspecified */
    KnownAreaUnit["Unspecified"] = "Unspecified";
    /** SquareKilometer */
    KnownAreaUnit["SquareKilometer"] = "SquareKilometer";
    /** SquareHectometer */
    KnownAreaUnit["SquareHectometer"] = "SquareHectometer";
    /** SquareDecameter */
    KnownAreaUnit["SquareDecameter"] = "SquareDecameter";
    /** SquareDecimeter */
    KnownAreaUnit["SquareDecimeter"] = "SquareDecimeter";
    /** SquareMeter */
    KnownAreaUnit["SquareMeter"] = "SquareMeter";
    /** SquareCentimeter */
    KnownAreaUnit["SquareCentimeter"] = "SquareCentimeter";
    /** SquareMillimeter */
    KnownAreaUnit["SquareMillimeter"] = "SquareMillimeter";
    /** SquareInch */
    KnownAreaUnit["SquareInch"] = "SquareInch";
    /** SquareFoot */
    KnownAreaUnit["SquareFoot"] = "SquareFoot";
    /** SquareMile */
    KnownAreaUnit["SquareMile"] = "SquareMile";
    /** SquareYard */
    KnownAreaUnit["SquareYard"] = "SquareYard";
    /** Acre */
    KnownAreaUnit["Acre"] = "Acre";
})(KnownAreaUnit || (KnownAreaUnit = {}));
/** Known values of {@link LengthUnit} that the service accepts. */
export var KnownLengthUnit;
(function (KnownLengthUnit) {
    /** Unspecified */
    KnownLengthUnit["Unspecified"] = "Unspecified";
    /** Kilometer */
    KnownLengthUnit["Kilometer"] = "Kilometer";
    /** Hectometer */
    KnownLengthUnit["Hectometer"] = "Hectometer";
    /** Decameter */
    KnownLengthUnit["Decameter"] = "Decameter";
    /** Meter */
    KnownLengthUnit["Meter"] = "Meter";
    /** Decimeter */
    KnownLengthUnit["Decimeter"] = "Decimeter";
    /** Centimeter */
    KnownLengthUnit["Centimeter"] = "Centimeter";
    /** Millimeter */
    KnownLengthUnit["Millimeter"] = "Millimeter";
    /** Micrometer */
    KnownLengthUnit["Micrometer"] = "Micrometer";
    /** Nanometer */
    KnownLengthUnit["Nanometer"] = "Nanometer";
    /** Picometer */
    KnownLengthUnit["Picometer"] = "Picometer";
    /** Mile */
    KnownLengthUnit["Mile"] = "Mile";
    /** Yard */
    KnownLengthUnit["Yard"] = "Yard";
    /** Inch */
    KnownLengthUnit["Inch"] = "Inch";
    /** Foot */
    KnownLengthUnit["Foot"] = "Foot";
    /** LightYear */
    KnownLengthUnit["LightYear"] = "LightYear";
    /** Pt */
    KnownLengthUnit["Pt"] = "Pt";
})(KnownLengthUnit || (KnownLengthUnit = {}));
/** Known values of {@link InformationUnit} that the service accepts. */
export var KnownInformationUnit;
(function (KnownInformationUnit) {
    /** Unspecified */
    KnownInformationUnit["Unspecified"] = "Unspecified";
    /** Bit */
    KnownInformationUnit["Bit"] = "Bit";
    /** Kilobit */
    KnownInformationUnit["Kilobit"] = "Kilobit";
    /** Megabit */
    KnownInformationUnit["Megabit"] = "Megabit";
    /** Gigabit */
    KnownInformationUnit["Gigabit"] = "Gigabit";
    /** Terabit */
    KnownInformationUnit["Terabit"] = "Terabit";
    /** Petabit */
    KnownInformationUnit["Petabit"] = "Petabit";
    /** Byte */
    KnownInformationUnit["Byte"] = "Byte";
    /** Kilobyte */
    KnownInformationUnit["Kilobyte"] = "Kilobyte";
    /** Megabyte */
    KnownInformationUnit["Megabyte"] = "Megabyte";
    /** Gigabyte */
    KnownInformationUnit["Gigabyte"] = "Gigabyte";
    /** Terabyte */
    KnownInformationUnit["Terabyte"] = "Terabyte";
    /** Petabyte */
    KnownInformationUnit["Petabyte"] = "Petabyte";
})(KnownInformationUnit || (KnownInformationUnit = {}));
/** Known values of {@link TemperatureUnit} that the service accepts. */
export var KnownTemperatureUnit;
(function (KnownTemperatureUnit) {
    /** Unspecified */
    KnownTemperatureUnit["Unspecified"] = "Unspecified";
    /** Fahrenheit */
    KnownTemperatureUnit["Fahrenheit"] = "Fahrenheit";
    /** Kelvin */
    KnownTemperatureUnit["Kelvin"] = "Kelvin";
    /** Rankine */
    KnownTemperatureUnit["Rankine"] = "Rankine";
    /** Celsius */
    KnownTemperatureUnit["Celsius"] = "Celsius";
})(KnownTemperatureUnit || (KnownTemperatureUnit = {}));
/** Known values of {@link WeightUnit} that the service accepts. */
export var KnownWeightUnit;
(function (KnownWeightUnit) {
    /** Unspecified */
    KnownWeightUnit["Unspecified"] = "Unspecified";
    /** Kilogram */
    KnownWeightUnit["Kilogram"] = "Kilogram";
    /** Gram */
    KnownWeightUnit["Gram"] = "Gram";
    /** Milligram */
    KnownWeightUnit["Milligram"] = "Milligram";
    /** Gallon */
    KnownWeightUnit["Gallon"] = "Gallon";
    /** MetricTon */
    KnownWeightUnit["MetricTon"] = "MetricTon";
    /** Ton */
    KnownWeightUnit["Ton"] = "Ton";
    /** Pound */
    KnownWeightUnit["Pound"] = "Pound";
    /** Ounce */
    KnownWeightUnit["Ounce"] = "Ounce";
    /** Grain */
    KnownWeightUnit["Grain"] = "Grain";
    /** PennyWeight */
    KnownWeightUnit["PennyWeight"] = "PennyWeight";
    /** LongTonBritish */
    KnownWeightUnit["LongTonBritish"] = "LongTonBritish";
    /** ShortTonUS */
    KnownWeightUnit["ShortTonUS"] = "ShortTonUS";
    /** ShortHundredWeightUS */
    KnownWeightUnit["ShortHundredWeightUS"] = "ShortHundredWeightUS";
    /** Stone */
    KnownWeightUnit["Stone"] = "Stone";
    /** Dram */
    KnownWeightUnit["Dram"] = "Dram";
})(KnownWeightUnit || (KnownWeightUnit = {}));
/** Known values of {@link DateTimeSubKind} that the service accepts. */
export var KnownDateTimeSubKind;
(function (KnownDateTimeSubKind) {
    /** Time */
    KnownDateTimeSubKind["Time"] = "Time";
    /** Date */
    KnownDateTimeSubKind["Date"] = "Date";
    /** DateTime */
    KnownDateTimeSubKind["DateTime"] = "DateTime";
    /** Duration */
    KnownDateTimeSubKind["Duration"] = "Duration";
    /** Set */
    KnownDateTimeSubKind["Set"] = "Set";
})(KnownDateTimeSubKind || (KnownDateTimeSubKind = {}));
/** Known values of {@link TemporalModifier} that the service accepts. */
export var KnownTemporalModifier;
(function (KnownTemporalModifier) {
    /** AfterApprox */
    KnownTemporalModifier["AfterApprox"] = "AfterApprox";
    /** Before */
    KnownTemporalModifier["Before"] = "Before";
    /** BeforeStart */
    KnownTemporalModifier["BeforeStart"] = "BeforeStart";
    /** Approx */
    KnownTemporalModifier["Approx"] = "Approx";
    /** ReferenceUndefined */
    KnownTemporalModifier["ReferenceUndefined"] = "ReferenceUndefined";
    /** SinceEnd */
    KnownTemporalModifier["SinceEnd"] = "SinceEnd";
    /** AfterMid */
    KnownTemporalModifier["AfterMid"] = "AfterMid";
    /** Start */
    KnownTemporalModifier["Start"] = "Start";
    /** After */
    KnownTemporalModifier["After"] = "After";
    /** BeforeEnd */
    KnownTemporalModifier["BeforeEnd"] = "BeforeEnd";
    /** Until */
    KnownTemporalModifier["Until"] = "Until";
    /** End */
    KnownTemporalModifier["End"] = "End";
    /** Less */
    KnownTemporalModifier["Less"] = "Less";
    /** Since */
    KnownTemporalModifier["Since"] = "Since";
    /** AfterStart */
    KnownTemporalModifier["AfterStart"] = "AfterStart";
    /** BeforeApprox */
    KnownTemporalModifier["BeforeApprox"] = "BeforeApprox";
    /** Mid */
    KnownTemporalModifier["Mid"] = "Mid";
    /** More */
    KnownTemporalModifier["More"] = "More";
})(KnownTemporalModifier || (KnownTemporalModifier = {}));
/** Known values of {@link NumberKind} that the service accepts. */
export var KnownNumberKind;
(function (KnownNumberKind) {
    /** Integer */
    KnownNumberKind["Integer"] = "Integer";
    /** Decimal */
    KnownNumberKind["Decimal"] = "Decimal";
    /** Power */
    KnownNumberKind["Power"] = "Power";
    /** Fraction */
    KnownNumberKind["Fraction"] = "Fraction";
    /** Percent */
    KnownNumberKind["Percent"] = "Percent";
    /** Unspecified */
    KnownNumberKind["Unspecified"] = "Unspecified";
})(KnownNumberKind || (KnownNumberKind = {}));
/** Known values of {@link RelativeTo} that the service accepts. */
export var KnownRelativeTo;
(function (KnownRelativeTo) {
    /** Current */
    KnownRelativeTo["Current"] = "Current";
    /** End */
    KnownRelativeTo["End"] = "End";
    /** Start */
    KnownRelativeTo["Start"] = "Start";
})(KnownRelativeTo || (KnownRelativeTo = {}));
/** Known values of {@link RangeKind} that the service accepts. */
export var KnownRangeKind;
(function (KnownRangeKind) {
    /** Number */
    KnownRangeKind["Number"] = "Number";
    /** Speed */
    KnownRangeKind["Speed"] = "Speed";
    /** Weight */
    KnownRangeKind["Weight"] = "Weight";
    /** Length */
    KnownRangeKind["Length"] = "Length";
    /** Volume */
    KnownRangeKind["Volume"] = "Volume";
    /** Area */
    KnownRangeKind["Area"] = "Area";
    /** Age */
    KnownRangeKind["Age"] = "Age";
    /** Information */
    KnownRangeKind["Information"] = "Information";
    /** Temperature */
    KnownRangeKind["Temperature"] = "Temperature";
    /** Currency */
    KnownRangeKind["Currency"] = "Currency";
})(KnownRangeKind || (KnownRangeKind = {}));
/** Known values of {@link ExtractiveSummarizationOrderingCriteria} that the service accepts. */
export var KnownExtractiveSummarizationOrderingCriteria;
(function (KnownExtractiveSummarizationOrderingCriteria) {
    /** Indicates that results should be sorted in order of appearance in the text. */
    KnownExtractiveSummarizationOrderingCriteria["Offset"] = "Offset";
    /** Indicates that results should be sorted in order of importance (i.e. rank score) according to the model. */
    KnownExtractiveSummarizationOrderingCriteria["Rank"] = "Rank";
})(KnownExtractiveSummarizationOrderingCriteria || (KnownExtractiveSummarizationOrderingCriteria = {}));
